javagen/src/main/java/com/azure/autorest/mapper/ModelMapper.java

```diff
@@ -433,7 +433,7 @@ private List<ClientModelPropertyReference> collectPropertiesFromFlattenedModel(
         ObjectSchema targetModelSchema = (ObjectSchema) property.getSchema();
         String originalFlattenedPropertyName = property.getLanguage().getJava().getName();  // not modelProperty.getName()
         ClientModel targetModel = this.map(targetModelSchema);
-        if (targetModel != null && !CoreUtils.isNullOrEmpty(targetModel.getProperties())) {
+        if (targetModel != null && targetModel.getProperties() != null) {
```

Comment:
`targetModel.getProperties() != null` seems redundant?

Reply:
technically not redundant, as `getProperties()` could be null?

though in reality if that be `null`, we probably just get lots of other places throwing NPE...